server/utils/scan_test.go

```diff
@@ -0,0 +1,189 @@
+package utils
+
+import (
+	"strings"
+	"testing"
+)
+
+// scanAlternate is an alternative to scan(). It is not currently faster in enough cases, but may be
+// faster in the future with improvements, or after more functions and interfaces have been added.
+func (ci *contractInfo) scanAlternate(byteCode string) {
+	if len(byteCode) < 8 {
+		return
+	}
+	ci.funcs = map[EVMFunction]struct{}{}
+	for i := 0; (i + 7) < len(byteCode); i++ {
+		id := byteCode[i : i+8]
+		if fn, ok := EVMFunctionsByID[id]; ok {
+			ci.funcs[fn] = struct{}{}
+		}
```

Comment:
I thought this would do better, since it's `O(N)` and the existing is `O(NxM)`, but the benchmark cases didn't show it.

Reply:
After the constant overheads stop dominating runtime, doubling the number of functions should double the runtime of the existing, but not affect the runtime of this version.